File: server/php/src/Application/Domain/Payments/PaymentDetails/ProtectionEligibility.php
<?php

namespace OnlinePayments\ExampleApp\Application\Domain\Payments\PaymentDetails;

final class ProtectionEligibility
{
    public ?string $eligibility = null;
    public ?string $type = null;

    public function toArray(): array
    {
        return [
            'eligibility' => $this->eligibility,
            'type' => $this->type,
        ];
    }
}

File: server/ruby/app/business/domain/payments/payment_details/protection_eligibility.rb
module Business
  module Domain
    module Payments
      module PaymentDetails
        class ProtectionEligibility
          attr_accessor :eligibility, :type

          def initialize(eligibility: nil, type: nil)
            @eligibility = eligibility
            @type = type
          end

          def to_h
            {
              'eligibility' => @eligibility,
              'type' => @type
            }
          end
        end
      end
    end
  end
end